### ISSUE TITLE
Cubestore: Allow STS to be used in AWS to assume role instead of only access key + secret key/instance metadata

### ISSUE THREAD

**Issue by keatmin** [cube-js/cube.js#5176]
**Is your feature request related to a problem? Please describe.**
Current implementation for S3 [credentials](https://github.com/cube-js/cube.js/blob/master/rust/cubestore/cubestore/src/remotefs/s3.rs#L54-L55) in Cubestore are only via long term access key + secret key pair or ec2 instance metadata. AWS users using EKS usually prefer to use short term temporary security credentials via STS to authenticate to their AWS services. 

**Describe the solution you'd like**
Adding a way for cube to use [this STS method](https://github.com/durch/rust-s3/blob/44f21c4d589f88b9d65fe2796cf48caea978b9eb/aws-creds/src/credentials.rs#L190) as a way to to obtain WebIdentityToken to authenticate to use S3.

**Comment by keatmin** [comment]
Hmm, closing for now as [it seems to be implemented in rust-s3 already](https://github.com/durch/rust-s3/blob/44f21c4d589f88b9d65fe2796cf48caea978b9eb/aws-creds/src/credentials.rs#L282-L285).

**Comment by Hunqer** [comment]
Hello @keatmin, 
is that working for you? Or cubestore require some code changes to use ARN_ROLE?

**Comment by booyaa** [comment]
Leaving this here for anyone else who comes across this issue. 

TL;DR You can't use assumed roles in Cube Store (not until they add support for it)

The S3 library that Cube Store depends on does allow the use of an assumed role, but this still requires implementation on Cube Store.  The current credentials handler only supports access key: https://github.com/cube-js/cube.js/blob/master/rust/cubestore/cubestore/src/remotefs/s3.rs#L54-L55

**Comment by keatmin** [comment]
@Hunqer yes it has been working for me. I was looking at the rust code of aws-creds, which is the library that is being used by cubestore and it seems like it [automatically](https://github.com/durch/rust-s3/blob/a513abc9dc6d9191f645efc5f722f84af85557bd/aws-creds/src/credentials.rs#L182) look for `sts_env` when you do `Credentials.new`. So as long as there's no env var that caused cubestore to be thinking that it's not something from env_var then it will work

**Comment by keatmin** [comment]
@booyaa I can assume role just fine with AWS_WEB_IDENTITY_TOKEN_FILE and AWS_ROLE_ARN via helm + EKS without the update. If you look into the rust-s3 code, it actually opt for STS before they use the access_key
